customer-commons/src/main/java/no/unit/nva/customer/model/CustomerDto.java

```diff
@@ -424,15 +423,9 @@ public Builder withSector(Sector sector) {
 
         private DoiAgentDto buildDoiAgentDto(DoiAgent doiAgent) {
             if (nonNull(doiAgent)) {
-                if (nonNull(customerDto.identifier)) {
-                    var doiAgentId =
-                        UriWrapper.fromUri(toId(customerDto.identifier))
-                            .addChild(DOI_AGENT)
-                            .getUri();
-                    return new DoiAgentDto(doiAgent)
-                               .addId(doiAgentId);
-                }
-                return new DoiAgentDto(doiAgent);
+                return nonNull(customerDto.identifier)
+                           ? new DoiAgentDto(doiAgent).addIdFromUuid(customerDto.identifier)
+                           : new DoiAgentDto(doiAgent);
```

Comment:
This no longer needs `.addId()`?

Reply:
Good question.
This will (should) not happen live, but some test scenarios does not have an identifier, so code to fix test introduced problems

Reply:
And I forgot, also for POST requests, where identifier hasn't been generated yet, when getting presisted, it will always have an identifier, and doiAgent.Id is constructed from indentifier, in the transformation from dao to dto